### PR TITLE
replace EthProvider/SolProvider P3A metrics, fix provider injection in existing renderers after creating wallet

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves https://github.com/brave/brave-browser/issues/30874
Resolves https://github.com/brave/brave-browser/issues/30585
Resolves https://github.com/brave/brave-browser/issues/30873

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Follow test plan for both `Brave.Wallet.EthProvider.4` and `Brave.Wallet.SolProvider.2`. Ensure metric responses are independent of each other.

0) Start with fresh profile. Access brave.com. Ensure metric has a value of 0.
1) Setup the wallet. Reload the brave.com page. Ensure metric has value of 2.
2) Access web3 settings and set the default provider to "Extensions (no fallback)". Reload the page. Ensure the metric has a value of 1.
3) Set the default provider setting to "brave wallet". Reload page. Ensure the metric has a value of 3.
4) Set the default provider setting to "extensions (brave wallet fallback)". Install & setup a third-party extension for the given chain. Reload page. Ensure the metric has a value of 5.
5) Set the default provider setting to "extensions (no fallback)". Reload page. Ensure the metric has a value of 4.
6) Set the default provider setting to "extensions (brave wallet fallback)", reset Brave Wallet. Reload page. Ensure metric has a value of 4.
7) Uninstall the third-party extension. Reload page. Ensure the metric has a value of 0.

### CODE REVIEW

**Review by iefremov — APPROVED** [review]
p3a lgtm

File: browser/profiles/brave_renderer_updater.h [hunk 1]
```diff
@@ -48,13 +58,15 @@ class BraveRendererUpdater : public KeyedService {
...
   raw_ptr<Profile> profile_ = nullptr;
+  raw_ptr<brave_wallet::KeyringService> keyring_service_ = nullptr;
   PrefChangeRegistrar pref_change_registrar_;
 
   // Prefs that we sync to the renderers.
   IntegerPrefMember brave_wallet_ethereum_provider_;
   IntegerPrefMember brave_wallet_solana_provider_;
   BooleanPrefMember de_amp_enabled_;
   bool is_wallet_allowed_for_context_;
+  bool is_wallet_created_;
```
Comment (supermassive) [inline browser/profiles/brave_renderer_updater.h:69]: This is not initialized in default ctor. Please prefer inline initiliazation like `is_wallet_created_ = false;`
and also same for `is_wallet_allowed_for_context_`

File: browser/profiles/brave_renderer_updater.cc [hunk 2]
```diff
@@ -38,6 +38,10 @@ BraveRendererUpdater::BraveRendererUpdater(Profile* profile)
   brave_wallet_solana_provider_.Init(kDefaultSolanaWallet, pref_service);
   de_amp_enabled_.Init(de_amp::kDeAmpPrefEnabled, pref_service);
 
+  keyring_service_ =
+      brave_wallet::KeyringServiceFactory::GetServiceForContext(profile);
+  CheckActiveWallet();
```
Comment (supermassive) [inline browser/profiles/brave_renderer_updater.cc:43]: - `BraveRendererUpdaterFactory` needs dependency(search for `DependsOn`) on `KeyringServiceFactory`
- `KeyringService` should come as an argument to `BraveRendererUpdater` ctor

File: components/brave_wallet/browser/brave_wallet_p3a.cc [hunk 3]
```diff
@@ -13,6 +13,7 @@
 #include "base/strings/string_number_conversions.h"
 #include "base/time/time.h"
 #include "brave/components/brave_wallet/browser/brave_wallet_service.h"
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
```
Comment (supermassive) [inline components/brave_wallet/browser/brave_wallet_p3a.cc:16]: need this include?

File: browser/profiles/brave_renderer_updater.cc (hunk 2 above)
Comment (DJAndries) [reply browser/profiles/brave_renderer_updater.cc:43]: good call, thanks

File: browser/profiles/brave_renderer_updater.cc [hunk 4]
```diff
@@ -31,13 +31,19 @@
...
+    Profile* profile,
+    brave_wallet::KeyringService* keyring_service)
+    : profile_(profile), keyring_service_(keyring_service) {
   PrefService* pref_service = profile->GetPrefs();
   brave_wallet_ethereum_provider_.Init(kDefaultEthereumWallet, pref_service);
   brave_wallet_solana_provider_.Init(kDefaultSolanaWallet, pref_service);
   de_amp_enabled_.Init(de_amp::kDeAmpPrefEnabled, pref_service);
 
+  keyring_service_ =
+      brave_wallet::KeyringServiceFactory::GetServiceForContext(profile);
```
Comment (supermassive) [inline browser/profiles/brave_renderer_updater.cc:44]: Don't need this now

**Review by supermassive — APPROVED** [review]
LGTM but please address this https://github.com/brave/brave-core/pull/18782#discussion_r1230398917

**Comment by GeetaSarvadnya** [comment]
Verification INPROGRESS on 

```
5790.32 (Official Build) nightly (64-bit)
-- | --
Revision | f9b78cf410ecd5b372310ef25a12ef456c77d90f-refs/branch-heads/5790@{#724}
OS | Windows 10 Version 22H2 (Build 19045.3086)

```

**Comment by GeetaSarvadnya** [comment]
Verification **INPROGRESS** on

```
Brave | 1.55.2 Chromium: 115.0.5790.32 (Official Build) nightly (64-bit)
-- | --
Revision | f9b78cf410ecd5b372310ef25a12ef456c77d90f-refs/branch-heads/5790@{#724}
OS | Windows 10 Version 22H2 (Build 19045.3086)
```

**Pull request merged**